src/Hosting/Hosting/src/Internal/HostingEnvironmentExtensions.cs

```diff
@@ -82,6 +82,9 @@ internal static void Initialize(this IWebHostEnvironment hostingEnvironment, str
             hostingEnvironment.ApplicationName = options.ApplicationName;
             hostingEnvironment.ContentRootPath = contentRootPath;
             hostingEnvironment.ContentRootFileProvider = new PhysicalFileProvider(hostingEnvironment.ContentRootPath);
+            //Defaulting here but it could be also interesting to do some work like checking if it does exists
+            //or creating the Directory if it doesn't.
+            hostingEnvironment.TempDirectoryPath = options.TempDirectory ?? Path.GetTempPath();
```

Comment:
You can use ??= in c# 8

Reply:
I'm not sure how to use ??= here. We are checking whether setting `hostingEnvironment.TempDirectoryPath` to `options.TempDirectory` (if it's defined via UseTempDirectory or with the env variable `ASPNETCORE_TEMP`) or defaulting to `Path.GetTempPath()`.

Reply:
Ah! Missed that